pkg/serdes/avro_serialization_provider.go

```diff
@@ -98,8 +125,16 @@ func (a *AvroSerializationProvider) Serialize(topic, message string) ([]byte, er
 		return nil, fmt.Errorf("failed to serialize message: %w", err)
 	}
 
-	// Step#4: Serialize the Go native data object with the confluent-kafka-go Serialize() library
-	payload, err := a.ser.Serialize(topic, &object)
+	// Step#4: Fetch the Go native data object, cast it into generic map for Serialize()
+	// Note: the suggested argument to pass to Serialize() library function is:
+	// - pointer to a generic map consistent with the schema during registration
+	// - a materialized object consistent with the schema during registration
+	// Passing the Go native object directly could cause issues during ruleSet execution
+	v, ok := object.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("failed to serialize message: unexpected message type asserion result")
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to serialize message: unexpected message type assertion result")
```
I didn't catch this on first pass

Reply:
No worry, I also happen to notice 2 typos, will update in a min.